tiny_dnn/io/caffe/layer_factory_impl.h

```diff
@@ -634,7 +648,7 @@ inline std::shared_ptr<layer> create_convlayer(
     }
 
     // 0 ... valid, (window_size-1)/2 ... same
-    if (pad_w == (window_size - 1) / 2) {
+    if (pad_w == (window_size_w - 1) / 2) {
```

Comment:
I think same padding works only if both width and hight of window matches the criteria. Something like ` window_size_w ==  (window_size_h && pad_w ==  (window_size_w - 1) / 2`